diff --git a/chrome/installer/linux/BUILD.gn b/chrome/installer/linux/BUILD.gn
index 1394af3..c3efe0d 100644
--- a/chrome/installer/linux/BUILD.gn
+++ b/chrome/installer/linux/BUILD.gn
@@ -84,9 +84,15 @@ copy("common_packaging_files") {
   ]
 
   if (is_chrome_branded) {
-    sources += [ "common/google-chrome/google-chrome.info" ]
+    sources += [
+      "common/google-chrome/google-chrome.appdata.xml.template",
+      "common/google-chrome/google-chrome.info"
+    ]
   } else {
-    sources += [ "common/chromium-browser/chromium-browser.info" ]
+    sources += [
+      "common/chromium-browser/chromium-browser.appdata.xml",
+      "common/chromium-browser/chromium-browser.info"
+    ]
   }
 
   if (current_cpu == "x86") {
diff --git a/chrome/installer/linux/common/chromium-browser/chromium-browser.appdata.xml b/chrome/installer/linux/common/chromium-browser/chromium-browser.appdata.xml
index e69de29..3bbe4f9 100644
--- a/chrome/installer/linux/common/chromium-browser/chromium-browser.appdata.xml
+++ b/chrome/installer/linux/common/chromium-browser/chromium-browser.appdata.xml
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>chromium-browser.desktop</id>
+  <update_contact>chromium-dev@chromium.org</update_contact>
+  <metadata_license>CC0-1.0 and (CC-BY-SA-3.0 or GFDL-1.2) and CC-BY-3.0 and CC-BY-SA-4.0</metadata_license>
+  <project_license>BSD-3-Clause and LGPL-2.1+ and Apache-2.0 and IJG and MIT and GPL-2.0+ and ISC and OpenSSL and (MPL-1.1 or GPL-2.0 or LGPL-2.0)</project_license>
+  <name>Chromium Web Browser</name>
+  <summary>Web Browser</summary>
+  <description>
+   <p>
+    Chromium is an open-source browser that aims to build a safer, faster, and
+    more stable way to experience the web.
+    We invite you to join us in our effort to build a powerful platform for
+    developing a new generation of web applications.
+   </p>
+   <p>
+    Chromium does not include the AAC, H.264, MP3 or Adobe Flash code
+    that is found in Chrome, although it does support Vorbis, Theora, WebM and
+    HTML5 audio and video standards.
+   </p>
+  </description>
+  <url type="homepage">https://www.chromium.org/Home</url>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://raw.githubusercontent.com/hughsie/fedora-appstream/master/screenshots-extra/chromium-browser/a.png</image>
+      <caption/>
+    </screenshot>
+    <screenshot>
+      <image>https://raw.githubusercontent.com/hughsie/fedora-appstream/master/screenshots-extra/chromium-browser/b.png</image>
+      <caption/>
+    </screenshot>
+  </screenshots>
+  <translation/>
+  <developer_name>The Chromium Authors</developer_name>
+  <url type="bugtracker">https://www.chromium.org/for-testers/bug-reporting-guidelines</url>
+  <url type="help">https://chromium.googlesource.com/chromium/src/+/master/docs/linux_debugging.md</url>
+</component>
diff --git a/chrome/installer/linux/common/google-chrome/google-chrome.appdata.xml.template b/chrome/installer/linux/common/google-chrome/google-chrome.appdata.xml.template
index e69de29..2c45632 100644
--- a/chrome/installer/linux/common/google-chrome/google-chrome.appdata.xml.template
+++ b/chrome/installer/linux/common/google-chrome/google-chrome.appdata.xml.template
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>@@PACKAGE@@.desktop</id>
+  <update_contact>chromium-dev@chromium.org</update_contact>
+  <metadata_license>CC0-1.0 and (CC-BY-SA-3.0 or GFDL-1.2) and CC-BY-3.0 and CC-BY-SA-4.0</metadata_license>
+  <project_license>Freeware under Google Chrome Terms of Service</project_license>
+  <name>@@MENUNAME@@</name>
+  <summary>Web Browser</summary>
+  <description>
+    <p>
+    @@FULLDESC@@
+   </p>
+  </description>
+  <url type="homepage">@@PRODUCTURL@@</url>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://raw.githubusercontent.com/hughsie/fedora-appstream/master/screenshots-extra/chromium-browser/a.png</image>
+      <caption/>
+    </screenshot>
+    <screenshot>
+      <image>https://raw.githubusercontent.com/hughsie/fedora-appstream/master/screenshots-extra/chromium-browser/b.png</image>
+      <caption/>
+    </screenshot>
+  </screenshots>
+  <translation/>
+  <developer_name>Google</developer_name>
+  <url type="bugtracker">https://support.google.com/chrome/answer/95315</url>
+  <url type="help">https://support.google.com/chrome</url>
+</component>
diff --git a/chrome/installer/linux/common/installer.include b/chrome/installer/linux/common/installer.include
index d606238..c2cb109 100644
--- a/chrome/installer/linux/common/installer.include
+++ b/chrome/installer/linux/common/installer.include
@@ -228,6 +228,14 @@ stage_install_common() {
   # desktop integration
   install -m 755 "${BUILDDIR}/xdg-mime" "${STAGEDIR}${INSTALLDIR}/"
   install -m 755 "${BUILDDIR}/xdg-settings" "${STAGEDIR}${INSTALLDIR}/"
+  if [ ${PACKAGE:0:6} = google ]; then
+    process_template "${BUILDDIR}/installer/common/google-chrome/google-chrome.appdata.xml.template" \
+      "${STAGEDIR}/usr/share/appdata/${PACKAGE}.appdata.xml"
+    chmod 644 "${STAGEDIR}/usr/share/appdata/${PACKAGE}.appdata.xml"
+  else
+    install -m 644 "${BUILDDIR}/installer/common/chromium-browser/chromium-browser.appdata.xml" \
+      "${STAGEDIR}/usr/share/appdata/${PACKAGE}.appdata.xml"
+  fi
   process_template "${BUILDDIR}/installer/common/desktop.template" \
     "${STAGEDIR}/usr/share/applications/${PACKAGE}.desktop"
   chmod 644 "${STAGEDIR}/usr/share/applications/${PACKAGE}.desktop"
diff --git a/chrome/installer/linux/rpm/chrome.spec.template b/chrome/installer/linux/rpm/chrome.spec.template
index 08c402c..e878a85 100644
--- a/chrome/installer/linux/rpm/chrome.spec.template
+++ b/chrome/installer/linux/rpm/chrome.spec.template
@@ -89,6 +89,7 @@ rm -rf "$RPM_BUILD_ROOT"
 /etc/cron.daily/@@PACKAGE@@
 %ghost %attr(755,root,root) /usr/bin/google-chrome
 /usr/bin/@@USR_BIN_SYMLINK_NAME@@
+/usr/share/appdata/@@PACKAGE@@.appdata.xml
 /usr/share/applications/@@PACKAGE@@.desktop
 /usr/share/gnome-control-center/default-apps/@@PACKAGE@@.xml
 %docdir /usr/share/man/man1
