diff -up chromium-75.0.3770.100/chrome/common/media_router/media_sink.cc.el7-noexcept chromium-75.0.3770.100/chrome/common/media_router/media_sink.cc
--- chromium-75.0.3770.100/chrome/common/media_router/media_sink.cc.el7-noexcept	2019-06-18 11:47:55.000000000 -0400
+++ chromium-75.0.3770.100/chrome/common/media_router/media_sink.cc	2019-06-20 10:17:27.440446569 -0400
@@ -19,12 +19,12 @@ MediaSink::MediaSink(const MediaSink::Id
       provider_id_(provider_id) {}
 
 MediaSink::MediaSink(const MediaSink& other) = default;
-MediaSink::MediaSink(MediaSink&& other) noexcept = default;
+MediaSink::MediaSink(MediaSink&& other) = default;
 MediaSink::MediaSink() = default;
 MediaSink::~MediaSink() = default;
 
 MediaSink& MediaSink::operator=(const MediaSink& other) = default;
-MediaSink& MediaSink::operator=(MediaSink&& other) noexcept = default;
+MediaSink& MediaSink::operator=(MediaSink&& other) = default;
 
 bool MediaSink::IsMaybeCloudSink() const {
   switch (icon_type_) {
diff -up chromium-75.0.3770.100/components/policy/core/common/policy_map.cc.el7-noexcept chromium-75.0.3770.100/components/policy/core/common/policy_map.cc
--- chromium-75.0.3770.100/components/policy/core/common/policy_map.cc.el7-noexcept	2019-06-18 11:48:01.000000000 -0400
+++ chromium-75.0.3770.100/components/policy/core/common/policy_map.cc	2019-06-20 10:17:27.444446463 -0400
@@ -31,7 +31,7 @@ PolicyMap::Entry::Entry(
 PolicyMap::Entry::~Entry() = default;
 
 PolicyMap::Entry::Entry(Entry&&) noexcept = default;
-PolicyMap::Entry& PolicyMap::Entry::operator=(Entry&&) noexcept = default;
+PolicyMap::Entry& PolicyMap::Entry::operator=(Entry&&) = default;
 
 PolicyMap::Entry PolicyMap::Entry::DeepCopy() const {
   Entry copy;
diff -up chromium-75.0.3770.100/components/signin/core/browser/account_info.cc.el7-noexcept chromium-75.0.3770.100/components/signin/core/browser/account_info.cc
--- chromium-75.0.3770.100/components/signin/core/browser/account_info.cc.el7-noexcept	2019-06-20 10:17:27.444446463 -0400
+++ chromium-75.0.3770.100/components/signin/core/browser/account_info.cc	2019-06-20 10:19:17.959532865 -0400
@@ -51,7 +51,7 @@ CoreAccountInfo::CoreAccountInfo(CoreAcc
 CoreAccountInfo& CoreAccountInfo::operator=(const CoreAccountInfo& other) =
     default;
 
-CoreAccountInfo& CoreAccountInfo::operator=(CoreAccountInfo&& other) noexcept =
+CoreAccountInfo& CoreAccountInfo::operator=(CoreAccountInfo&& other) =
     default;
 
 bool CoreAccountInfo::IsEmpty() const {
diff -up chromium-75.0.3770.100/gpu/config/gpu_info.cc.el7-noexcept chromium-75.0.3770.100/gpu/config/gpu_info.cc
--- chromium-75.0.3770.100/gpu/config/gpu_info.cc.el7-noexcept	2019-06-18 11:48:10.000000000 -0400
+++ chromium-75.0.3770.100/gpu/config/gpu_info.cc	2019-06-20 10:17:27.445446437 -0400
@@ -181,7 +181,7 @@ GPUInfo::GPUDevice& GPUInfo::GPUDevice::
     const GPUInfo::GPUDevice& other) = default;
 
 GPUInfo::GPUDevice& GPUInfo::GPUDevice::operator=(
-    GPUInfo::GPUDevice&& other) noexcept = default;
+    GPUInfo::GPUDevice&& other) = default;
 
 GPUInfo::GPUInfo()
     : optimus(false),
