diff -up chromium-76.0.3809.100/chrome/common/media_router/media_sink.cc.el7-noexcept chromium-76.0.3809.100/chrome/common/media_router/media_sink.cc
--- chromium-76.0.3809.100/chrome/common/media_router/media_sink.cc.el7-noexcept	2019-08-23 13:41:04.842597850 +0200
+++ chromium-76.0.3809.100/chrome/common/media_router/media_sink.cc	2019-08-23 13:41:04.842597850 +0200
@@ -19,12 +19,12 @@ MediaSink::MediaSink(const MediaSink::Id
       provider_id_(provider_id) {}
 
 MediaSink::MediaSink(const MediaSink& other) = default;
-MediaSink::MediaSink(MediaSink&& other) noexcept = default;
+MediaSink::MediaSink(MediaSink&& other) = default;
 MediaSink::MediaSink() = default;
 MediaSink::~MediaSink() = default;
 
 MediaSink& MediaSink::operator=(const MediaSink& other) = default;
-MediaSink& MediaSink::operator=(MediaSink&& other) noexcept = default;
+MediaSink& MediaSink::operator=(MediaSink&& other) = default;
 
 bool MediaSink::IsMaybeCloudSink() const {
   switch (icon_type_) {
diff -up chromium-76.0.3809.100/components/history/core/browser/history_types.cc.el7-noexcept chromium-76.0.3809.100/components/history/core/browser/history_types.cc
--- chromium-76.0.3809.100/components/history/core/browser/history_types.cc.el7-noexcept	2019-08-23 13:45:57.673815531 +0200
+++ chromium-76.0.3809.100/components/history/core/browser/history_types.cc	2019-08-26 10:13:00.815409764 +0200
@@ -42,7 +42,7 @@ QueryResults::QueryResults(QueryResults&
   Swap(&other);
 }
 
-QueryResults& QueryResults::operator=(QueryResults&& other) noexcept {
+QueryResults& QueryResults::operator=(QueryResults&& other) {
   Swap(&other);
   return *this;
 }
@@ -186,7 +186,7 @@ QueryURLResult::QueryURLResult(QueryURLR
 
 QueryURLResult& QueryURLResult::operator=(const QueryURLResult&) = default;
 
-QueryURLResult& QueryURLResult::operator=(QueryURLResult&&) noexcept = default;
+QueryURLResult& QueryURLResult::operator=(QueryURLResult&&) = default;
 
 // MostVisitedURL --------------------------------------------------------------
 
diff -up chromium-76.0.3809.100/components/history/core/browser/history_types.h.el7-noexcept chromium-76.0.3809.100/components/history/core/browser/history_types.h
--- chromium-76.0.3809.100/components/history/core/browser/history_types.h.el7-noexcept	2019-08-23 13:46:11.455966965 +0200
+++ chromium-76.0.3809.100/components/history/core/browser/history_types.h	2019-08-26 10:13:49.951881558 +0200
@@ -143,7 +143,7 @@ class QueryResults {
   ~QueryResults();
 
   QueryResults(QueryResults&& other) noexcept;
-  QueryResults& operator=(QueryResults&& other) noexcept;
+  QueryResults& operator=(QueryResults&& other);
 
   void set_reached_beginning(bool reached) { reached_beginning_ = reached; }
   bool reached_beginning() { return reached_beginning_; }
@@ -278,7 +278,7 @@ struct QueryURLResult {
   QueryURLResult(const QueryURLResult&);
   QueryURLResult(QueryURLResult&&) noexcept;
   QueryURLResult& operator=(const QueryURLResult&);
-  QueryURLResult& operator=(QueryURLResult&&) noexcept;
+  QueryURLResult& operator=(QueryURLResult&&);
   ~QueryURLResult();
 
   // Indicates whether the call to HistoryBackend::QueryURL was successfull
diff -up chromium-76.0.3809.100/components/policy/core/common/policy_map.cc.el7-noexcept chromium-76.0.3809.100/components/policy/core/common/policy_map.cc
--- chromium-76.0.3809.100/components/policy/core/common/policy_map.cc.el7-noexcept	2019-08-23 13:40:42.633353810 +0200
+++ chromium-76.0.3809.100/components/policy/core/common/policy_map.cc	2019-08-23 13:40:42.576353183 +0200
@@ -52,7 +52,7 @@ PolicyMap::Entry::Entry(
 PolicyMap::Entry::~Entry() = default;
 
 PolicyMap::Entry::Entry(Entry&&) noexcept = default;
-PolicyMap::Entry& PolicyMap::Entry::operator=(Entry&&) noexcept = default;
+PolicyMap::Entry& PolicyMap::Entry::operator=(Entry&&) = default;
 
 PolicyMap::Entry PolicyMap::Entry::DeepCopy() const {
   Entry copy;
diff -up chromium-76.0.3809.100/components/signin/core/browser/account_info.cc.el7-noexcept chromium-76.0.3809.100/components/signin/core/browser/account_info.cc
--- chromium-76.0.3809.100/components/signin/core/browser/account_info.cc.el7-noexcept	2019-08-23 13:40:44.224371292 +0200
+++ chromium-76.0.3809.100/components/signin/core/browser/account_info.cc	2019-08-23 13:40:44.190370918 +0200
@@ -52,7 +52,7 @@ CoreAccountInfo::CoreAccountInfo(CoreAcc
 CoreAccountInfo& CoreAccountInfo::operator=(const CoreAccountInfo& other) =
     default;
 
-CoreAccountInfo& CoreAccountInfo::operator=(CoreAccountInfo&& other) noexcept =
+CoreAccountInfo& CoreAccountInfo::operator=(CoreAccountInfo&& other) =
     default;
 
 bool CoreAccountInfo::IsEmpty() const {
@@ -69,7 +69,7 @@ AccountInfo::AccountInfo(AccountInfo&& o
 
 AccountInfo& AccountInfo::operator=(const AccountInfo& other) = default;
 
-AccountInfo& AccountInfo::operator=(AccountInfo&& other) noexcept = default;
+AccountInfo& AccountInfo::operator=(AccountInfo&& other) = default;
 
 bool AccountInfo::IsEmpty() const {
   return CoreAccountInfo::IsEmpty() && hosted_domain.empty() &&
diff -up chromium-76.0.3809.100/google_apis/gaia/core_account_id.cc.el7-noexcept chromium-76.0.3809.100/google_apis/gaia/core_account_id.cc
--- chromium-76.0.3809.100/google_apis/gaia/core_account_id.cc.el7-noexcept	2019-08-23 13:40:25.468165195 +0200
+++ chromium-76.0.3809.100/google_apis/gaia/core_account_id.cc	2019-08-23 13:40:25.481165338 +0200
@@ -14,7 +14,7 @@ CoreAccountId::~CoreAccountId() = defaul
 
 CoreAccountId& CoreAccountId::operator=(const CoreAccountId&) = default;
 
-CoreAccountId& CoreAccountId::operator=(CoreAccountId&&) noexcept = default;
+CoreAccountId& CoreAccountId::operator=(CoreAccountId&&) = default;
 
 CoreAccountId::CoreAccountId(const char* id) : id(id) {}
 
diff -up chromium-76.0.3809.100/google_apis/gaia/core_account_id.h.el7-noexcept chromium-76.0.3809.100/google_apis/gaia/core_account_id.h
--- chromium-76.0.3809.100/google_apis/gaia/core_account_id.h.el7-noexcept	2019-08-23 13:40:25.468165195 +0200
+++ chromium-76.0.3809.100/google_apis/gaia/core_account_id.h	2019-08-23 13:40:25.481165338 +0200
@@ -19,7 +19,7 @@ struct CoreAccountId {
   ~CoreAccountId();
 
   CoreAccountId& operator=(const CoreAccountId&);
-  CoreAccountId& operator=(CoreAccountId&&) noexcept;
+  CoreAccountId& operator=(CoreAccountId&&);
 
   // Those implicit constructor and conversion operator allow to
   // progressively migrate the code to use this struct. Removing
diff -up chromium-76.0.3809.100/gpu/config/gpu_info.cc.el7-noexcept chromium-76.0.3809.100/gpu/config/gpu_info.cc
--- chromium-76.0.3809.100/gpu/config/gpu_info.cc.el7-noexcept	2019-08-09 16:48:01.000000000 +0200
+++ chromium-76.0.3809.100/gpu/config/gpu_info.cc	2019-08-23 13:40:25.482165349 +0200
@@ -166,7 +166,7 @@ GPUInfo::GPUDevice& GPUInfo::GPUDevice::
     const GPUInfo::GPUDevice& other) = default;
 
 GPUInfo::GPUDevice& GPUInfo::GPUDevice::operator=(
-    GPUInfo::GPUDevice&& other) noexcept = default;
+    GPUInfo::GPUDevice&& other) = default;
 
 GPUInfo::GPUInfo()
     : optimus(false),
