diff -up chromium-78.0.3904.70/base/logging.h.constexpr chromium-78.0.3904.70/base/logging.h
--- chromium-78.0.3904.70/base/logging.h.constexpr	2019-10-23 10:06:53.434313793 -0400
+++ chromium-78.0.3904.70/base/logging.h	2019-10-23 10:07:17.490853038 -0400
@@ -530,7 +530,7 @@ BASE_EXPORT extern std::ostream* g_swall
 class CheckOpResult {
  public:
   // |message| must be non-null if and only if the check failed.
-  CheckOpResult(std::string* message) : message_(message) {}
+  constexpr CheckOpResult(std::string* message) : message_(message) {}
   // Returns true if the check succeeded.
   operator bool() const { return !message_; }
   // Returns the message.
