diff -up chromium-70.0.3538.77/sandbox/linux/system_headers/arm64_linux_syscalls.h.aarch64-new-stat chromium-70.0.3538.77/sandbox/linux/system_headers/arm64_linux_syscalls.h
--- chromium-70.0.3538.77/sandbox/linux/system_headers/arm64_linux_syscalls.h.aarch64-new-stat	2018-11-06 13:27:05.118766581 -0500
+++ chromium-70.0.3538.77/sandbox/linux/system_headers/arm64_linux_syscalls.h	2018-11-06 13:27:34.575204504 -0500
@@ -5,6 +5,8 @@
 #ifndef SANDBOX_LINUX_SYSTEM_HEADERS_ARM64_LINUX_SYSCALLS_H_
 #define SANDBOX_LINUX_SYSTEM_HEADERS_ARM64_LINUX_SYSCALLS_H_
 
+#define __ARCH_WANT_NEW_STAT
+
 #include <asm-generic/unistd.h>
 
 #if !defined(__NR_io_setup)
