diff -up chromium-71.0.3578.98/chrome/common/media_router/media_sink.cc.fix-noexcept chromium-71.0.3578.98/chrome/common/media_router/media_sink.cc
--- chromium-71.0.3578.98/chrome/common/media_router/media_sink.cc.fix-noexcept	2019-01-02 14:44:40.449801789 -0500
+++ chromium-71.0.3578.98/chrome/common/media_router/media_sink.cc	2019-01-02 14:44:54.167481936 -0500
@@ -24,7 +24,7 @@ MediaSink::MediaSink() = default;
 MediaSink::~MediaSink() = default;
 
 MediaSink& MediaSink::operator=(const MediaSink& other) = default;
-MediaSink& MediaSink::operator=(MediaSink&& other) noexcept = default;
+MediaSink& MediaSink::operator=(MediaSink&& other) = default;
 
 bool MediaSink::Equals(const MediaSink& other) const {
   return sink_id_ == other.sink_id_;
diff -up chromium-71.0.3578.98/components/policy/core/common/policy_map.cc.fix-noexcept chromium-71.0.3578.98/components/policy/core/common/policy_map.cc
--- chromium-71.0.3578.98/components/policy/core/common/policy_map.cc.fix-noexcept	2019-01-02 12:06:10.867313533 -0500
+++ chromium-71.0.3578.98/components/policy/core/common/policy_map.cc	2019-01-02 12:06:27.544980078 -0500
@@ -18,7 +18,7 @@ PolicyMap::Entry::Entry() = default;
 PolicyMap::Entry::~Entry() = default;
 
 PolicyMap::Entry::Entry(Entry&&) noexcept = default;
-PolicyMap::Entry& PolicyMap::Entry::operator=(Entry&&) noexcept = default;
+PolicyMap::Entry& PolicyMap::Entry::operator=(Entry&&) = default;
 
 PolicyMap::Entry PolicyMap::Entry::DeepCopy() const {
   Entry copy;
diff -up chromium-71.0.3578.98/components/signin/core/browser/account_info.cc.fix-noexcept chromium-71.0.3578.98/components/signin/core/browser/account_info.cc
--- chromium-71.0.3578.98/components/signin/core/browser/account_info.cc.fix-noexcept	2019-01-02 10:57:29.194710662 -0500
+++ chromium-71.0.3578.98/components/signin/core/browser/account_info.cc	2019-01-02 10:58:57.829965474 -0500
@@ -32,7 +32,7 @@ AccountInfo::AccountInfo(AccountInfo&& o
 
 AccountInfo& AccountInfo::operator=(const AccountInfo& other) = default;
 
-AccountInfo& AccountInfo::operator=(AccountInfo&& other) noexcept = default;
+AccountInfo& AccountInfo::operator=(AccountInfo&& other) = default;
 
 bool AccountInfo::IsEmpty() const {
   return account_id.empty() && email.empty() && gaia.empty() &&
diff -up chromium-71.0.3578.98/gpu/config/gpu_info.cc.fix-noexcept chromium-71.0.3578.98/gpu/config/gpu_info.cc
--- chromium-71.0.3578.98/gpu/config/gpu_info.cc.fix-noexcept	2018-12-20 09:50:03.320449807 -0500
+++ chromium-71.0.3578.98/gpu/config/gpu_info.cc	2018-12-20 09:50:25.768849091 -0500
@@ -116,7 +116,7 @@ GPUInfo::GPUDevice& GPUInfo::GPUDevice::
     const GPUInfo::GPUDevice& other) = default;
 
 GPUInfo::GPUDevice& GPUInfo::GPUDevice::operator=(
-    GPUInfo::GPUDevice&& other) noexcept = default;
+    GPUInfo::GPUDevice&& other) = default;
 
 GPUInfo::GPUInfo()
     : optimus(false),
