diff -up chromium-57.0.2987.21/third_party/ffmpeg/libavutil/cpu.c.ffmpeg_stdatomic chromium-57.0.2987.21/third_party/ffmpeg/libavutil/cpu.c
--- chromium-57.0.2987.21/third_party/ffmpeg/libavutil/cpu.c.ffmpeg_stdatomic	2017-02-03 07:07:36.000000000 +0100
+++ chromium-57.0.2987.21/third_party/ffmpeg/libavutil/cpu.c	2017-02-09 11:54:34.083000442 +0100
@@ -20,6 +20,10 @@
 // Chromium: Windows doesn't provide stdatomic.h, so use the compat version.
 #if defined(_MSC_VER)
 #include <compat/atomics/win32/stdatomic.h>
+// GCC 4.8 didn't have stdatomic, but was advertising it.
+// https://gcc.gnu.org/bugzilla/show_bug.cgi?id=58016
+#elif !defined(__clang__) && defined(__GNUC__) && (__GNUC__ == 4 || (__GNUC__ == 4 && (__GNUC_MINOR__ == 8)))
+#include <compat/atomics/gcc/stdatomic.h>
 #else
 #include <stdatomic.h>
 #endif
