diff -up chromium-61.0.3163.79/cc/paint/BUILD.gn.epel7 chromium-61.0.3163.79/cc/paint/BUILD.gn
--- chromium-61.0.3163.79/cc/paint/BUILD.gn.epel7	2017-09-13 13:21:07.743006338 -0400
+++ chromium-61.0.3163.79/cc/paint/BUILD.gn	2017-09-13 13:25:43.091933209 -0400
@@ -41,6 +41,8 @@ cc_component("paint") {
     "solid_color_analyzer.h",
   ]
 
+  cflags = [ "-fpermissive" ]
+
   defines = [ "CC_PAINT_IMPLEMENTATION=1" ]
 
   # cc/paint is intended to be a separate component from cc that can be
