diff -up chromium-66.0.3359.117/services/preferences/tracked/pref_hash_filter.h.gcc-full-decl chromium-66.0.3359.117/services/preferences/tracked/pref_hash_filter.h
--- chromium-66.0.3359.117/services/preferences/tracked/pref_hash_filter.h.gcc-full-decl	2018-04-25 16:29:52.372692179 -0400
+++ chromium-66.0.3359.117/services/preferences/tracked/pref_hash_filter.h	2018-04-25 16:31:15.698058409 -0400
@@ -21,9 +21,9 @@
 #include "services/preferences/public/mojom/preferences.mojom.h"
 #include "services/preferences/tracked/hash_store_contents.h"
 #include "services/preferences/tracked/interceptable_pref_filter.h"
+#include "services/preferences/tracked/pref_hash_store.h"
 #include "services/preferences/tracked/tracked_preference.h"
 
-class PrefHashStore;
 class PrefService;
 
 namespace base {
diff -up chromium-66.0.3359.117/third_party/WebKit/Source/modules/webdatabase/SQLTransactionBackend.h.gcc-full-decl chromium-66.0.3359.117/third_party/WebKit/Source/modules/webdatabase/SQLTransactionBackend.h
--- chromium-66.0.3359.117/third_party/WebKit/Source/modules/webdatabase/SQLTransactionBackend.h.gcc-full-decl	2018-04-17 21:05:12.000000000 -0400
+++ chromium-66.0.3359.117/third_party/WebKit/Source/modules/webdatabase/SQLTransactionBackend.h	2018-04-25 16:29:52.373692159 -0400
@@ -30,6 +30,7 @@
 
 #include <memory>
 #include "modules/webdatabase/DatabaseBasicTypes.h"
+#include "modules/webdatabase/SQLError.h"
 #include "modules/webdatabase/SQLStatement.h"
 #include "modules/webdatabase/SQLStatementBackend.h"
 #include "modules/webdatabase/SQLTransactionStateMachine.h"
@@ -41,7 +42,6 @@
 namespace blink {
 
 class Database;
-class SQLErrorData;
 class SQLiteTransaction;
 class SQLTransaction;
 class SQLTransactionBackend;
