diff -up chromium-74.0.3729.169/base/values.h.gentoo3 chromium-74.0.3729.169/base/values.h
--- chromium-74.0.3729.169/base/values.h.gentoo3	2019-05-31 14:02:13.019800039 -0400
+++ chromium-74.0.3729.169/base/values.h	2019-05-31 14:02:32.531380130 -0400
@@ -84,7 +84,7 @@ class BASE_EXPORT Value {
   using DictStorage = flat_map<std::string, std::unique_ptr<Value>>;
   using ListStorage = std::vector<Value>;
 
-  enum class Type {
+  enum class Type : unsigned char {
     NONE = 0,
     BOOLEAN,
     INTEGER,
