diff -up chromium-73.0.3683.86/sandbox/linux/seccomp-bpf-helpers/syscall_parameters_restrictions.cc.clonevfork chromium-73.0.3683.86/sandbox/linux/seccomp-bpf-helpers/syscall_parameters_restrictions.cc
--- chromium-73.0.3683.86/sandbox/linux/seccomp-bpf-helpers/syscall_parameters_restrictions.cc.clonevfork	2019-04-11 10:22:21.250929060 -0400
+++ chromium-73.0.3683.86/sandbox/linux/seccomp-bpf-helpers/syscall_parameters_restrictions.cc	2019-04-11 10:23:58.832770803 -0400
@@ -146,7 +146,8 @@ ResultExpr RestrictCloneToThreadsAndEPER
 
   const uint64_t kGlibcPthreadFlags =
       CLONE_VM | CLONE_FS | CLONE_FILES | CLONE_SIGHAND | CLONE_THREAD |
-      CLONE_SYSVSEM | CLONE_SETTLS | CLONE_PARENT_SETTID | CLONE_CHILD_CLEARTID;
+      CLONE_SYSVSEM | CLONE_SETTLS | CLONE_PARENT_SETTID |
+      CLONE_CHILD_CLEARTID | CLONE_VFORK;
   const BoolExpr glibc_test = flags == kGlibcPthreadFlags;
 
   const BoolExpr android_test =
