diff -up chromium-63.0.3239.84/third_party/webrtc/p2p/base/port.cc.gcc-round-fix chromium-63.0.3239.84/third_party/webrtc/p2p/base/port.cc
--- chromium-63.0.3239.84/third_party/webrtc/p2p/base/port.cc.gcc-round-fix	2017-12-07 16:20:01.521717091 -0500
+++ chromium-63.0.3239.84/third_party/webrtc/p2p/base/port.cc	2017-12-07 16:20:13.025491739 -0500
@@ -11,6 +11,7 @@
 #include "p2p/base/port.h"
 
 #include <algorithm>
+#include <cmath>
 #include <vector>
 
 #include "p2p/base/common.h"
