diff -up chromium-72.0.3626.121/chrome/common/safe_browsing/BUILD.gn.nounrar chromium-72.0.3626.121/chrome/common/safe_browsing/BUILD.gn
--- chromium-72.0.3626.121/chrome/common/safe_browsing/BUILD.gn.nounrar	2019-03-10 14:15:47.187705953 -0400
+++ chromium-72.0.3626.121/chrome/common/safe_browsing/BUILD.gn	2019-03-10 14:16:47.154138649 -0400
@@ -56,40 +56,6 @@ if (safe_browsing_mode == 1) {
     ]
   }
 
-  source_set("rar_analyzer") {
-    sources = [
-      "rar_analyzer.cc",
-      "rar_analyzer.h",
-    ]
-
-    deps = [
-      ":archive_analyzer_results",
-      ":download_type_util",
-      ":file_type_policies",
-      "//base",
-      "//base:i18n",
-      "//third_party/unrar:unrar",
-    ]
-
-    defines = [
-      "_FILE_OFFSET_BITS=64",
-      "LARGEFILE_SOURCE",
-      "RAR_SMP",
-      "SILENT",
-
-      # The following is set to disable certain macro definitions in the unrar
-      # source code.
-      "CHROMIUM_UNRAR",
-
-      # Disables exceptions in unrar, replaces them with process termination.
-      "UNRAR_NO_EXCEPTIONS",
-    ]
-
-    public_deps = [
-      "//components/safe_browsing:csd_proto",
-    ]
-  }
-
   source_set("disk_image_type_sniffer_mac") {
     sources = [
       "disk_image_type_sniffer_mac.cc",
@@ -150,7 +116,6 @@ source_set("safe_browsing") {
     deps += [
       ":archive_analyzer_results",
       ":download_type_util",
-      ":rar_analyzer",
       "//components/safe_browsing:features",
     ]
 
diff -up chromium-72.0.3626.121/chrome/common/safe_browsing/DEPS.nounrar chromium-72.0.3626.121/chrome/common/safe_browsing/DEPS
--- chromium-72.0.3626.121/chrome/common/safe_browsing/DEPS.nounrar	2019-03-01 23:10:03.000000000 -0500
+++ chromium-72.0.3626.121/chrome/common/safe_browsing/DEPS	2019-03-10 14:14:55.430058716 -0400
@@ -1,6 +1,5 @@
 include_rules = [
   "+components/safe_browsing",
   "+third_party/protobuf",
-  "+third_party/unrar",
   "+third_party/zlib",
 ]
diff -up chromium-72.0.3626.121/chrome/services/file_util/safe_archive_analyzer.cc.nounrar chromium-72.0.3626.121/chrome/services/file_util/safe_archive_analyzer.cc
--- chromium-72.0.3626.121/chrome/services/file_util/safe_archive_analyzer.cc.nounrar	2019-03-01 23:10:04.000000000 -0500
+++ chromium-72.0.3626.121/chrome/services/file_util/safe_archive_analyzer.cc	2019-03-10 14:14:55.433058638 -0400
@@ -47,9 +47,13 @@ void SafeArchiveAnalyzer::AnalyzeDmgFile
 
 void SafeArchiveAnalyzer::AnalyzeRarFile(base::File rar_file,
                                          AnalyzeRarFileCallback callback) {
+#if 0
   DCHECK(rar_file.IsValid());
 
   safe_browsing::ArchiveAnalyzerResults results;
   safe_browsing::rar_analyzer::AnalyzeRarFile(std::move(rar_file), &results);
   std::move(callback).Run(results);
+#else
+  NOTREACHED();
+#endif
 }
