diff -up chromium-76.0.3809.132/chrome/test/chromedriver/BUILD.gn.fixme chromium-76.0.3809.132/chrome/test/chromedriver/BUILD.gn
--- chromium-76.0.3809.132/chrome/test/chromedriver/BUILD.gn.fixme	2019-09-10 15:53:02.871463568 -0400
+++ chromium-76.0.3809.132/chrome/test/chromedriver/BUILD.gn	2019-09-10 16:57:32.281112323 -0400
@@ -319,6 +319,7 @@ executable("chromedriver") {
     "//net/server:http_server",
     "//net/traffic_annotation:test_support",
     "//services/network/public/mojom",
+    "//third_party/zlib",
   ]
 }
 
