diff -up chromium-77.0.3865.75/content/common/user_agent.cc.fedora-user-agent chromium-77.0.3865.75/content/common/user_agent.cc
--- chromium-77.0.3865.75/content/common/user_agent.cc.fedora-user-agent	2019-09-12 15:49:11.902270729 +0200
+++ chromium-77.0.3865.75/content/common/user_agent.cc	2019-09-12 15:50:11.555732044 +0200
@@ -35,7 +35,7 @@ std::string GetUserAgentPlatform() {
 #elif defined(OS_MACOSX)
   return "Macintosh; ";
 #elif defined(USE_X11) || defined(USE_OZONE)
-  return "X11; ";  // strange, but that's what Firefox uses
+  return "X11; Fedora; ";  // strange, but that's what Firefox uses
 #elif defined(OS_ANDROID)
   return "Linux; ";
 #elif defined(OS_POSIX)
